src/displayComponents/addToCartButton/addToCart.js

```diff
@@ -0,0 +1,18 @@
+import "../../styles/button.css";
+import {useState} from "react";
+import { useCartContext } from "../../context/cartcontext";
+import { ADD_TO_CART } from "../../constants/constants";
+function ButtonAddToCart({item}){
+    const [inCartStatus,setInCartStatus]=useState(false);
+    const {dispatch}=useCartContext();
+    return(
+        <button class="btn-text btn-style-color btn" onClick={()=>{
+                dispatch({type:ADD_TO_CART,item});
+                setInCartStatus((inCartStatus)=>!inCartStatus);              
```

Comment:
Extract the functionality out to a separate function
i.e 
```
const addtoCart = () => {
  dispatch({type: ADD_TO_CART, payload: item})
  setInCartStatus((inCartStatus)=>!inCartStatus);
}
 onClick={() => addtoCart()}
 
```
And please pass data to reducer as payload